src/opnsense/www/js/widgets/DnsmasqLeases.js

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2025 Deciso B.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+export default class DnsmasqLeases extends BaseTableWidget {
+
+    constructor(config) {
+        super(config);
+        this.configurable = true;
+        this.configChanged = false;
+    }
+
+    getGridOptions() {
+        return {
+            sizeToContent: 650
+        };
+    }
+
+    getMarkup() {
+        return this.createTable('dnsmasqLeasesTable', {
+            headerPosition: 'left'
+        });
+    }
+
+    async onWidgetTick() {
+        $('[data-toggle="tooltip"]').tooltip({container: 'body'});
+
+        const settingsResponse = await this.ajaxCall(`/api/dnsmasq/settings/${'get'}`);
+        if (settingsResponse?.dnsmasq?.enable !== '1') {
+            this.displayError(this.translations.unconfigured, '/ui/dnsmasq/settings#general');
+            return;
+        }
+
+        const leasesResponse = await this.ajaxCall(`/api/dnsmasq/leases/${'search'}`);
```

Comment:
we can probably push the `limit` logic back to the controller (`POST` + `rowCount` property)